deb/debian/stackdriver-agent.init

```diff
@@ -120,7 +121,7 @@ start() {
     fi
 
     LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/opt/stackdriver/collectd/lib64:/opt/stackdriver/collectd/lib" \
-        start_daemon -p "$_PIDFILE" "$DAEMON" -C "$CONFIG" -P "$_PIDFILE" \
+        start_daemon -p "$_PIDFILE" "$COLLECTDMON" -P "$_PIDFILE" -c "$DAEMON" -- -C "$CONFIG" \
```

Comment:
Should probably also change `stop()` and the `status`/`condrestart` cases below, since we're no longer looking for `$DAEMON`… Also in `rpm/stackdriver-agent`.

Reply:
To be consistent, i have just changed it all to $COLLECTDMON. 

In the meanwhile, i notice for those places that were using "$DAEMON", e.x. `pidofproc -p "$_PIDFILE" "$DAEMON" >/dev/null` it doesn't matter what you have for the daemon name, I put a non-sense word, so that it looks like this: `pidofproc -p "$_PIDFILE" "asdfasdfasdfasdf" >/dev/null`. Running commands: 
sudo systemctl daemon-reload
sudo /etc/init.d/stackdriver-agent status (i have set -x in the init.d)
--->
```
+++ rc=0
+++ '[' status = status ']'
+++ return 0
+++ exit 0
```

So i think it only cares about the pidfile name. The same stays true for `killproc -p "$_PIDFILE" "$DAEMON"` too.

Reply:
Another place that could be debatable on which daemon name to use is check_config, i think we shall use $DAEMON. See below:
sophieyfang@debian9-meow:~$ LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/opt/stackdriver/collectd/lib64:/opt/stackdriver/collectd/lib" "/opt/stackdriver/collectd/sbin/stackdriver-collectd" -t -C "/etc/stackdriver/collectd.conf"
sophieyfang@debian9-meow:~$ echo $?
0
sophieyfang@debian9-meow:~$ LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/opt/stackdriver/collectd/lib64:/opt/stackdriver/collectd/lib" "/opt/stackdriver/collectd/sbin/stackdriver-collectdmon" -t -C "/etc/stackdriver/collectd.conf"
/opt/stackdriver/collectd/sbin/stackdriver-collectdmon: invalid option -- 't'
Usage: /opt/stackdriver/collectd/sbin/stackdriver-collectdmon <options> [-- <collectd options>]

Available options:
  -h         Display this help and exit.
  -c <path>  Path to the collectd binary.
  -P <file>  PID-file.

For <collectd options> see collectd.conf(5).

collectd 5.8.1.git, http://collectd.org/
by Florian octo Forster <octo@collectd.org>
for contributions see `AUTHORS'
sophieyfang@debian9-meow:~$ echo $?
0

Reply:
Note that above two comments, those statements are based on tests inside debian9, it might not be representative for all distros

Reply:
Right, for testing the configuration we have to use `$DAEMON`, as that runs collectd in the foreground.

Regarding the argument to `pidofproc`, it only matters if the pid file is missing or corrupted, so it has a process name to fall back on. What you have now looks correct.